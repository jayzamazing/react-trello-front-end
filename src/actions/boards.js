import {normalize} from 'normalizr';
import {boardArray, boardsSchema} from '../board-schema';
import {BASE_URL} from '../config';

/*
* action to tell store that all boards has been retrieved
* @params boards - boards to be sent to store
* @returns action type and boards
*/
export const FIND_BOARDS_SUCCESS = 'FIND_BOARDS_SUCCESS';
export const findBoardsSuccess = boards => {
  //grab all boards array with cards and cardslist and normalize it
  const items = (normalize(boards, boardArray)).entities;
  return {
    type: FIND_BOARDS_SUCCESS,
    items
  };
};
/*
* function to get all boards
* @params findBoardsSuccess or passed in action
* @dispatch findBoardsSuccess or passed in action
*/
export const getBoards = (action = findBoardsSuccess) => dispatch => {
  return fetch(`${BASE_URL}/boards`)
    .then((res) => {
      if (!res.ok) return Promise.reject(res.statusText);
      dispatch(action(res.body));
    });
};
/*
* action to tell store that a board has been created
* @params boards - boards to be sent to store
* @returns action type and boards
*/
export const CREATE_BOARD_SUCCESS = 'CREATE_BOARD_SUCCESS';
export const createBoardSuccess = boards => {
  const items = (normalize(boards, boardsSchema)).entities;
  return {
    type: CREATE_BOARD_SUCCESS,
    items
  };
};
/*
* function to create a board
* @params postData - boards to update on the board
* @params createBoardSuccess or passed in action
* @dispatch createBoardSuccess or passed in action
*/
export const createBoards = (postData, action = createBoardSuccess) => dispatch => {
  return fetch(`${BASE_URL}/boards`, {
    method: "POST",
    body: postData
  })
    .then((res) => {
      if (!res.ok) return Promise.reject(res.statusText);
      dispatch(action(res.body));
    });
};
/*
* action to tell store that a board has been deleted
* @params id - id of the board to delete in the store
* @returns action type and boards
*/
export const DELETE_BOARD_SUCCESS = 'DELETE_BOARD_SUCCESS';
export const deleteBoardSuccess = id => {
  return {
    type: DELETE_BOARD_SUCCESS,
    boardId: id
  };
};
/*
* function to delete a board
* @params id - id of the board to delete
* @params deleteBoardSuccess or passed in action
* @dispatch deleteBoardSuccess or passed in action
*/
export const deleteBoards = (id, action = deleteBoardSuccess) => dispatch => {
  return fetch(`${BASE_URL}/boards/${id}`, {
    method: 'DELETE'
  })
    .then((res) => {
      if (!res.ok) return Promise.reject(res.statusText);
      dispatch(action(id));
    });
};
/*
* action to tell store that a board has been updated
* @params boards - boards to be sent to store
* @params id - id of the board to update in the store
* @returns action type and boards
*/
export const UPDATE_BOARD_SUCCESS = 'UPDATE_BOARD_SUCCESS';
export const updateBoardSuccess = function(id, boards) {
  const items = (normalize(boards, boardsSchema)).entities;
  return {
    type: UPDATE_BOARD_SUCCESS,
    items,
    boardId: id
  };
};
/*
* function to update the board
* @params id - id of the board to update
* @params postData - boards to update on the board
* @params updateBoardSuccess or passed in action
* @dispatch updateBoardSuccess or passed in action
*/
export const updateBoards = (id, postData, action = updateBoardSuccess) => dispatch => {
  return fetch(`${BASE_URL}/boards/${id}`, {
    method: 'PUT',
    body: postData
  })
    .then((res) => {
      if (!res.ok) return Promise.reject(res.statusText);
      dispatch(action(id, postData));
    });
};
