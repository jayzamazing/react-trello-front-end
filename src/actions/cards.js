import {normalize} from 'normalizr';
import {cardsSchema} from '../board-schema';
import {BASE_URL} from '../config';
/*
* action to tell store that a cards has been created
* @params data - data to be sent to store
* @returns action type and data
*/
export const CREATE_CARDS_SUCCESS = 'CREATE_CARDS_SUCCESS';
export const createCardsSuccess = cards => {
  const items = (normalize(cards, cardsSchema)).entities;
  return {
    type: CREATE_CARDS_SUCCESS,
    items
  };
};
/*
* function to create a cards
* @params postData - data to update on the cards
* @params createCardsSuccess or passed in action
* @dispatch createCardsSuccess or passed in action
*/
export const createCards = (postData, action = createCardsSuccess) => dispatch => {
  return fetch(`${BASE_URL}/cards`, {
    method: "POST",
    body: postData
  })
    .then(res => {
      if (!res.ok) return Promise.reject(res.statusText);
      return dispatch(action(res.body));
    });
};
/*
* action to tell store that a cards has been created
* @params data - data to be sent to store
* @returns action type and data
*/
export const DELETE_CARDS_SUCCESS = 'DELETE_CARDS_SUCCESS';
export const deleteCardsSuccess = id => {
  return {
    type: DELETE_CARDS_SUCCESS,
    cardsId: id
  };
};
/*
* function to delete a cards
* @params id - id of the cards to delete
* @params deleteCardsSuccess or passed in action
* @dispatch deleteCardsSuccess or passed in action
*/
export const deleteCards = (id, action = deleteCardsSuccess) => dispatch => {
  return fetch(`${BASE_URL}/cards/${id}`, {
    method: 'DELETE'
  })
    .then((res) => {
      if (!res.ok) return Promise.reject(res.statusText);
      dispatch(action(id));
    });
};
/*
* action to tell store that a cards has been updated
* @params cards - cards to be sent to store
* @params id - id of the cards to update in the store
* @returns action type and cards
*/
export const UPDATE_CARDS_SUCCESS = 'UPDATE_CARDS_SUCCESS';
export const updateCardsSuccess = function(id, cards) {
  const items = (normalize(cards, cardsSchema)).entities;
  return {
    type: UPDATE_CARDS_SUCCESS,
    items,
    cardsId: id
  };
};
/*
* function to update the cards
* @params id - id of the cards to update
* @params postData - cards to update on the board
* @params updateCardsSuccess or passed in action
* @dispatch updateCardsSuccess or passed in action
*/
export const updateCards = (id, postData, action = updateCardsSuccess) => dispatch => {
  return fetch(`${BASE_URL}/cards/${id}`, {
    method: 'PUT',
    body: postData
  })
    .then((res) => {
      if (!res.ok) return Promise.reject(res.statusText);
      dispatch(action(id, postData));
    });
};
