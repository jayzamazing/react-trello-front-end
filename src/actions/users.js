import {SubmissionError} from 'redux-form';
import {BASE_URL} from '../config';
import {normalizeResponseErrors} from './utils';

export const createUser = user => dispatch => {
  return fetch(`${BASE_URL}/users`, {
    method: 'POST',
    headers: {
      'content-type': 'application/json'
    },
    body: JSON.stringify(user)
  }).then(res => normalizeResponseErrors(res))
  .then(res => res.json())
  .catch(err => {
    const {reason, message, location} = err;
    return Promise.reject(
      return Promise.reject(
        new SubmissionError({
          [location]: message
        });;
      );
    );
  });
}
