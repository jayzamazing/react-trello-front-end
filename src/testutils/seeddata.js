'use strict';
import faker from 'faker';
//used to create a single board in json
export const seedBoardsSingle = (title = faker.random.words(), _id = faker.random.alphaNumeric(20)) => {
  return {
    title: title,
    _id: _id,
    cardsList: seedCardslistIds()
  };
};
//used to create random boards based on count
export const seedBoardsCount = (count) => {
  if (!count) {
    count = 1;
  }
  let boards = [];
  for (var i = 0; i < count; i++) {
    let _id = faker.random.alphaNumeric(20);
    boards.push({
      _id: _id,
      title: faker.random.words(),
      cardsList: seedCardslistIds(count)
    });
  }
  return boards;
}
//create board items based on input values, after normalization
export const seedBoardsInput = (ids, words, ids2) => {
  let boards = {};
  for (var i = 0; i < ids.length; i++) {
    var _id = ids[i];
    boards[_id] = {
      _id: _id,
      title: words[i],
      cardslist: ids2[i]
    };
  }
  return boards;
}
//used to create cardslist ids for seedboards
export const seedCardslistIds = (count) => {
  let cardslistIds = [];
  for (var i = 0; i < count; i++) {
    cardslistIds.push(faker.random.alphaNumeric(20));
  }
  return cardslistIds;
}
//used to create a single cardslist in json
export const seedCardslistSingle = (boardId = faker.random.alphaNumeric(20), title = faker.random.words(), _id = faker.random.alphaNumeric(20)) => {
  return {
    boardId: boardId,
    title: title,
    _id: _id,
    cards: seedCardsIds()
  };
}
//used to create random cardslist based on count
export const seedCardslistCount = (boardId, count) => {
  if (!count) {
    count = 1;
  }
  let cardslist = [];
  for (var i = 0; i < count; i++) {
    cardslist.push({
      boardId: boardId,
      _id: faker.random.alphaNumeric(20),
      title: faker.random.words(),
      cards: seedCardsIds(count)
    });
  }
  return cardslist;
}
//create cardslist items based on input value, after normalization
export const seedCardslistInput = function(boardId, words, ids) {
  let cardslist = {};
  for (var i = 0; i < ids.length; i++) {
    var _id = ids[i];
    cardslist[_id]= {
      boardId: boardId[i],
      _id: _id,
      title: words[i],
      cards: seedCardsIds(ids.length)
    };
  }
  return cardslist;
};
//used to create cardslist ids for seedboards
export const seedCardsIds = (count) => {
  let cardsIds = [];
  for (var i = 0; i < count; i++) {
    cardsIds.push(faker.random.alphaNumeric(20));
  }
  return cardsIds;
}
//used to create a single card in json
export const seedCardsSingle = (cardslistId = faker.random.alphaNumeric(20), text = faker.random.words(), _id = faker.random.alphaNumeric(20)) => {
  return {
    cardslistId: cardslistId,
    text: text,
    _id: _id
  };
}
//used to create random cardslist based on count
export const seedCardsCount = (cardslistId, count) => {
  if (!count) {
    count = 1;
  }
  let cards = [];
  for (var i = 0; i < count; i++) {
    cards.push({
      cardslistId: cardslistId,
      _id: faker.random.alphaNumeric(20),
      text: faker.random.words()
    });
  }
  return cards;
}
//create cards items based on input value, after normalization
export const seedCardsInput = function(cardslistId, words, ids) {
  let cards = {};
  for (var i = 0; i < ids.length; i++) {
    var _id = faker.random.alphaNumeric(20);
    cards[_id]= {
      cardslistId: cardslistId[i],
      _id: _id,
      text: faker.random.words()
    };
  }
  return cards;
};
//create a random title
export const createTitle = () => {
return {title: faker.random.words()};
};
