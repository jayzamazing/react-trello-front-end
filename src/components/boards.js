import React from 'react';
import {connect} from 'react-redux';
import CreateItems from './create-items';
import * as actions from '../actions/boards';
import Immutable from 'seamless-immutable';
import {Redirect} from 'react-router-dom';
//function to render multiple lists of boards
export class Boards extends React.Component {
  //set up initial data state
  constructor(props) {
    super(props);
    this.state = {
      showCreateBoard: false,
      editBoard: {},
      boards: {},
      boardTitle: ''
    }
  }
  componentDidMount() {
    if (!this.props.loggedIn) {
        return;
    }
    this.props.getBoards();
  }
  //keep track of text
  onAddInputChanged(event) {
    //if the addBoard input is being used
    if (event.target.name === 'addBoard') {
      //store title for board
      this.setState({boardTitle: event.target.value});
      //otherwise assume we are editing board name
    } else {
      //get boards from state
      var temp = this.state.boards;
      //update the title for the selected board
      var temp2 = Immutable.update(temp,
        event.target.id,
        function() {
          return {
            title: event.target.value
          };
        });
      //store the updated board title
      this.setState({boards: temp2});
    }
  }
  //hide create board when called
  addBoard() {
    this.setState({showCreateBoard: false});
  }
  //set the variable to show the create board inputs
  showCreateBoard() {
    this.setState({showCreateBoard: true});
  }
  //set variable to enable the editing of the boards name
  editBoardName(item) {
    var temp = this.state.editBoard;
    temp[item] = false;
    this.setState({editBoard: temp});
  }
  showBoard(boardId, boardName, item) {
    if (this.state.editBoard[item] === undefined) {
      var temp = this.state.editBoard;
      temp[item] = true;
      this.setState({editBoard: temp});
    }
    if (this.state.editBoard[item] === true) {
      this.props.history.push('/:' + boardId + '/:' + boardName);
    }
  }
  render() {
    // Only visible to logged in users
    if (!this.props.loggedIn) {
        return <Redirect to="/" />;
    }
    console.log(this.props);
    //only execute if there is data
    if (this.props.boards) {
      var list = Object.keys(this.props.boards).map((item, index) => {
        var temp = this.props.boards[item];
        return (
          <li key={index}>
            <span onClick={() => this.showBoard(null, temp._id, temp.title, temp._id)}>
              <input type="text" id={temp._id} value={this.state.boards[temp._id] ? this.state.boards[temp._id].title : temp.title}
                disabled={(this.state.editBoard[temp._id] === undefined) ? true : this.state.editBoard[temp._id] }
                onChange={(evt) => this.onAddInputChanged(evt)}
                onKeyPress={(evt) => this.props.updateBoard(evt)} name="boardName"/>
            </span>
            <input type="button" value="Delete Board" name="deleteBoard"
              onClick={() => this.props.deleteBoard(temp._id)}/>
            <input type="button" value="Edit Board" name="editBoard"
              onClick={() => this.editBoardName(temp._id)}/>
          </li>
        );
      });
    }
    return (
      <div>
        <ul>{list}</ul>
        <input type="button" value="Add Board" onClick={() => this.showCreateBoard()} name="addBoard"/>
        {this.state.showCreateBoard ?
          <CreateItems onAddInputChanged={(evt) => this.onAddInputChanged(evt)}
            addItems={() => {this.addBoard(); this.props.addBoard();}} name="boardInput"/> : null}
      </div>
    );
  }
};
//allows subcription to redux updates and access to data stored in redux store
const mapStateToProps = (state, props) => {
  const {currentUser} = state.auth;
  return {
    boards: state.boards,
    loggedIn: currentUser !== null,
    username: currentUser ? state.auth.currentUser.username : '',
    fullName: currentUser
        ? `${currentUser.fullName}`
        : ''
  };
};
const mapDispatchToProps = (dispatch, props) => ({
    //get all boards
    getBoards: () => {
      dispatch(actions.getBoards())
    },
    //dispatch to delete board
    deleteBoard: (boardId) => {
      dispatch(actions.deleteBoards(boardId));
    },
    //dispatch to add board
    addBoard: () => {
      dispatch(actions.createBoards({title: props.boardTitle}));
    },
    //dispatch update to board name if enter key is press in field
    updateBoard: (evt) => {
      if(evt.charCode === 13) {
        dispatch(actions.updateBoards(evt.target.id, {title: evt.target.value}));
      }
    }
});
export default connect(mapStateToProps, mapDispatchToProps)(Boards);
