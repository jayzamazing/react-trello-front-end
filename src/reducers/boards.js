import Immutable from 'seamless-immutable';
import {FIND_BOARDS_SUCCESS, CREATE_BOARD_SUCCESS, DELETE_BOARD_SUCCESS, UPDATE_BOARD_SUCCESS} from '../actions/boards';


const initialRepositoryState = Immutable({});

/*
* Function called when DELETE_BOARD_SUCCESS is called to deal with deleting a board and removing it from state
* params state- old state before merge
* params action.boardid- board to delete
*/
export const deleteBoard = (state, action) => (state, {boardId}) => state.without(boardId);

/*
* Function called when UPDATE_BOARD_SUCCESS is called to update boards with a new board
* params state- old state before merge
* params action.items.boards- boards to update
*/
export const updateBoard = (state, {boards}) => state.merge(boards, {deep: true});

/*
* Function to deal with using various reducer functions
* params state- old state before merge
* params action- action with data to modify state
*/
export default (state = initialRepositoryState, action) => {
  switch (action.type) {
  case FIND_BOARDS_SUCCESS:
  case CREATE_BOARD_SUCCESS:
  case UPDATE_BOARD_SUCCESS:
    return updateBoard(state, action);
  case DELETE_BOARD_SUCCESS:
    return deleteBoard(state, action);
  default:
    return state;
  }
};
