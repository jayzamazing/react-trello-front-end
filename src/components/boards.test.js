'use strict';
import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow, mount} from 'enzyme';
import component, {Boards} from './boards';
import {seedBoardsInput} from '../testutils/seeddata';
import * as actions from '../actions/boards';
import renderer from 'react-test-renderer';


//mock actions
const mockFindBoards = {
  type: 'FIND_BOARDS'
};
const mockDeleteBoards = {
  type: 'DELETE_BOARDS'
};
const mockUpdateBoards = {
  type: 'UPDATE_BOARDS'
};
//setup mock api
jest.mock('../actions/boards', () => Object.assign({},
require.requireActual('../actions/boards'),
{
  getBoards: jest.fn().mockImplementation(() => {
    return mockFindBoards;
  }),
  //needed for true comparison
  deleteBoards: jest.fn().mockImplementation(() => {
    return mockDeleteBoards;
  }),
  //needed for true comparison
  updateBoards: jest.fn().mockImplementation(() => {
    return mockUpdateBoards;
  })
}
));
//testing for the boards class
describe('Boards component', () => {
  let boards = {};
  //setup before all tests
  beforeEach(() => {
    let ids = ['2skdde39hzdo08aqcdgd', '14w43d4opubteu021ato', 'kd5o3ve2858vpjxd7wjp'];
    let words = ['XML Health', 'Bedfordshire Illinois Liaison', 'Sports Analyst Cambridgeshire'];
    boards = seedBoardsInput(ids, words);
  });
  //tear down after tests are complete
  afterEach(() => {
    boards = {};
  });
  //ensure board is rendered without issues
  it('should render', () => {
    const dispatch = jest.fn();
    shallow(<Boards boards={boards} dispatch={dispatch}/>);
  });
  //render board and check against the previous snapshot
  it('boards snapshot', () => {
    const dispatch = jest.fn();
    const wrapper = shallow(<Boards boards={boards} dispatch={dispatch}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  //click on add button and check before and after snapshots
  it('add board click snapshot', () => {
    const dispatch = jest.fn();
    const wrapper = mount(<Boards boards={boards} dispatch={dispatch}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.find('[name="addBoard"]').simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  //check dispatch is fired off when delete button is clicked
  it('dispatches delete board', () => {
    const deleteBoard = jest.fn();
    const dispatch = jest.fn();
    const wrapper = shallow(<Boards boards={boards} dispatch={dispatch} deleteBoard={deleteBoard}/>);
    const keys = Object.keys(boards);
    wrapper.find('[name="deleteBoard"]').at(0).simulate('click');
    expect(deleteBoard).toHaveBeenCalledWith(keys[0]);
  });
  //click on delete button and check before and after snapshots
  it('delete board snapshot', () => {
    const deleteBoard = jest.fn();
    const dispatch = jest.fn();
    const wrapper = mount(<Boards boards={boards} dispatch={dispatch} deleteBoard={deleteBoard}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.find('[name="deleteBoard"]').at(2).simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  //check dispatch is fired when update button is clicked
  it('dispatches update board', () => {
    const updateBoard = jest.fn();
    const wrapper = shallow(<Boards boards={boards} updateBoard={updateBoard}/>);
    const keys = Object.keys(boards);
    wrapper.find('[name="boardName"]').at(0).simulate('keypress', {key: 'Enter'})
    expect(updateBoard).toHaveBeenCalledWith({"key": "Enter"});
  });
  //click on update button and check before and after snapshots
  it('update board snapshot', () => {
    const updateBoard = jest.fn();
    const wrapper = mount(<Boards boards={boards} updateBoard={updateBoard}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.find('[name="editBoard"]').at(0).simulate('click');
    const keys = Object.keys(boards);
    wrapper.find('[name="boardName"]').at(0).simulate('change', { target: { value: 'testing', id: keys[0] } });
    wrapper.find('[name="boardName"]').at(0).simulate('keypress', {key: 'Enter'})
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
