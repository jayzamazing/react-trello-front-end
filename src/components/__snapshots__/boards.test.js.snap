// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Boards component add board click snapshot 1`] = `
<Boards
  boards={Array []}
  dispatch={[Function]}
>
  <div>
    <ul>
      <li
        key="0"
      >
        <span
          onClick={[Function]}
        >
          <input
            disabled={true}
            id="2skdde39hzdo08aqcdgd"
            name="boardName"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="text"
            value="XML Health"
          />
        </span>
        <input
          name="deleteBoard"
          onClick={[Function]}
          type="button"
          value="Delete Board"
        />
        <input
          name="editBoard"
          onClick={[Function]}
          type="button"
          value="Edit Board"
        />
      </li>
      <li
        key="1"
      >
        <span
          onClick={[Function]}
        >
          <input
            disabled={true}
            id="14w43d4opubteu021ato"
            name="boardName"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="text"
            value="Bedfordshire Illinois Liaison"
          />
        </span>
        <input
          name="deleteBoard"
          onClick={[Function]}
          type="button"
          value="Delete Board"
        />
        <input
          name="editBoard"
          onClick={[Function]}
          type="button"
          value="Edit Board"
        />
      </li>
      <li
        key="2"
      >
        <span
          onClick={[Function]}
        >
          <input
            disabled={true}
            id="kd5o3ve2858vpjxd7wjp"
            name="boardName"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="text"
            value="Sports Analyst Cambridgeshire"
          />
        </span>
        <input
          name="deleteBoard"
          onClick={[Function]}
          type="button"
          value="Delete Board"
        />
        <input
          name="editBoard"
          onClick={[Function]}
          type="button"
          value="Edit Board"
        />
      </li>
    </ul>
    <input
      name="addBoard"
      onClick={[Function]}
      type="button"
      value="Add Board"
    />
  </div>
</Boards>
`;

exports[`Boards component add board click snapshot 2`] = `
<Boards
  boards={Array []}
  dispatch={[Function]}
>
  <div>
    <ul>
      <li
        key="0"
      >
        <span
          onClick={[Function]}
        >
          <input
            disabled={true}
            id="2skdde39hzdo08aqcdgd"
            name="boardName"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="text"
            value="XML Health"
          />
        </span>
        <input
          name="deleteBoard"
          onClick={[Function]}
          type="button"
          value="Delete Board"
        />
        <input
          name="editBoard"
          onClick={[Function]}
          type="button"
          value="Edit Board"
        />
      </li>
      <li
        key="1"
      >
        <span
          onClick={[Function]}
        >
          <input
            disabled={true}
            id="14w43d4opubteu021ato"
            name="boardName"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="text"
            value="Bedfordshire Illinois Liaison"
          />
        </span>
        <input
          name="deleteBoard"
          onClick={[Function]}
          type="button"
          value="Delete Board"
        />
        <input
          name="editBoard"
          onClick={[Function]}
          type="button"
          value="Edit Board"
        />
      </li>
      <li
        key="2"
      >
        <span
          onClick={[Function]}
        >
          <input
            disabled={true}
            id="kd5o3ve2858vpjxd7wjp"
            name="boardName"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="text"
            value="Sports Analyst Cambridgeshire"
          />
        </span>
        <input
          name="deleteBoard"
          onClick={[Function]}
          type="button"
          value="Delete Board"
        />
        <input
          name="editBoard"
          onClick={[Function]}
          type="button"
          value="Edit Board"
        />
      </li>
    </ul>
    <input
      name="addBoard"
      onClick={[Function]}
      type="button"
      value="Add Board"
    />
    <CreateItems
      addItems={[Function]}
      name="boardInput"
      onAddInputChanged={[Function]}
    >
      <form
        className="list-form"
      >
        <input
          name="boardInput"
          onChange={[Function]}
          type="text"
        />
        <button
          onClick={[Function]}
          type="submit"
          value="Submit"
        />
      </form>
    </CreateItems>
  </div>
</Boards>
`;

exports[`Boards component boards snapshot 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <span
        onClick={[Function]}
      >
        <input
          disabled={true}
          id="2skdde39hzdo08aqcdgd"
          name="boardName"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value="XML Health"
        />
      </span>
      <input
        name="deleteBoard"
        onClick={[Function]}
        type="button"
        value="Delete Board"
      />
      <input
        name="editBoard"
        onClick={[Function]}
        type="button"
        value="Edit Board"
      />
    </li>
    <li
      key="1"
    >
      <span
        onClick={[Function]}
      >
        <input
          disabled={true}
          id="14w43d4opubteu021ato"
          name="boardName"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value="Bedfordshire Illinois Liaison"
        />
      </span>
      <input
        name="deleteBoard"
        onClick={[Function]}
        type="button"
        value="Delete Board"
      />
      <input
        name="editBoard"
        onClick={[Function]}
        type="button"
        value="Edit Board"
      />
    </li>
    <li
      key="2"
    >
      <span
        onClick={[Function]}
      >
        <input
          disabled={true}
          id="kd5o3ve2858vpjxd7wjp"
          name="boardName"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value="Sports Analyst Cambridgeshire"
        />
      </span>
      <input
        name="deleteBoard"
        onClick={[Function]}
        type="button"
        value="Delete Board"
      />
      <input
        name="editBoard"
        onClick={[Function]}
        type="button"
        value="Edit Board"
      />
    </li>
  </ul>
  <input
    name="addBoard"
    onClick={[Function]}
    type="button"
    value="Add Board"
  />
</div>
`;

exports[`Boards component delete board snapshot 1`] = `
<Boards
  boards={Array []}
  deleteBoard={[Function]}
  dispatch={[Function]}
>
  <div>
    <ul>
      <li
        key="0"
      >
        <span
          onClick={[Function]}
        >
          <input
            disabled={true}
            id="2skdde39hzdo08aqcdgd"
            name="boardName"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="text"
            value="XML Health"
          />
        </span>
        <input
          name="deleteBoard"
          onClick={[Function]}
          type="button"
          value="Delete Board"
        />
        <input
          name="editBoard"
          onClick={[Function]}
          type="button"
          value="Edit Board"
        />
      </li>
      <li
        key="1"
      >
        <span
          onClick={[Function]}
        >
          <input
            disabled={true}
            id="14w43d4opubteu021ato"
            name="boardName"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="text"
            value="Bedfordshire Illinois Liaison"
          />
        </span>
        <input
          name="deleteBoard"
          onClick={[Function]}
          type="button"
          value="Delete Board"
        />
        <input
          name="editBoard"
          onClick={[Function]}
          type="button"
          value="Edit Board"
        />
      </li>
      <li
        key="2"
      >
        <span
          onClick={[Function]}
        >
          <input
            disabled={true}
            id="kd5o3ve2858vpjxd7wjp"
            name="boardName"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="text"
            value="Sports Analyst Cambridgeshire"
          />
        </span>
        <input
          name="deleteBoard"
          onClick={[Function]}
          type="button"
          value="Delete Board"
        />
        <input
          name="editBoard"
          onClick={[Function]}
          type="button"
          value="Edit Board"
        />
      </li>
    </ul>
    <input
      name="addBoard"
      onClick={[Function]}
      type="button"
      value="Add Board"
    />
  </div>
</Boards>
`;

exports[`Boards component delete board snapshot 2`] = `
<Boards
  boards={Array []}
  deleteBoard={[Function]}
  dispatch={[Function]}
>
  <div>
    <ul>
      <li
        key="0"
      >
        <span
          onClick={[Function]}
        >
          <input
            disabled={true}
            id="2skdde39hzdo08aqcdgd"
            name="boardName"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="text"
            value="XML Health"
          />
        </span>
        <input
          name="deleteBoard"
          onClick={[Function]}
          type="button"
          value="Delete Board"
        />
        <input
          name="editBoard"
          onClick={[Function]}
          type="button"
          value="Edit Board"
        />
      </li>
      <li
        key="1"
      >
        <span
          onClick={[Function]}
        >
          <input
            disabled={true}
            id="14w43d4opubteu021ato"
            name="boardName"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="text"
            value="Bedfordshire Illinois Liaison"
          />
        </span>
        <input
          name="deleteBoard"
          onClick={[Function]}
          type="button"
          value="Delete Board"
        />
        <input
          name="editBoard"
          onClick={[Function]}
          type="button"
          value="Edit Board"
        />
      </li>
      <li
        key="2"
      >
        <span
          onClick={[Function]}
        >
          <input
            disabled={true}
            id="kd5o3ve2858vpjxd7wjp"
            name="boardName"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="text"
            value="Sports Analyst Cambridgeshire"
          />
        </span>
        <input
          name="deleteBoard"
          onClick={[Function]}
          type="button"
          value="Delete Board"
        />
        <input
          name="editBoard"
          onClick={[Function]}
          type="button"
          value="Edit Board"
        />
      </li>
    </ul>
    <input
      name="addBoard"
      onClick={[Function]}
      type="button"
      value="Add Board"
    />
  </div>
</Boards>
`;

exports[`Boards component update board snapshot 1`] = `
<Boards
  boards={Array []}
  updateBoard={[Function]}
>
  <div>
    <ul>
      <li
        key="0"
      >
        <span
          onClick={[Function]}
        >
          <input
            disabled={true}
            id="2skdde39hzdo08aqcdgd"
            name="boardName"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="text"
            value="XML Health"
          />
        </span>
        <input
          name="deleteBoard"
          onClick={[Function]}
          type="button"
          value="Delete Board"
        />
        <input
          name="editBoard"
          onClick={[Function]}
          type="button"
          value="Edit Board"
        />
      </li>
      <li
        key="1"
      >
        <span
          onClick={[Function]}
        >
          <input
            disabled={true}
            id="14w43d4opubteu021ato"
            name="boardName"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="text"
            value="Bedfordshire Illinois Liaison"
          />
        </span>
        <input
          name="deleteBoard"
          onClick={[Function]}
          type="button"
          value="Delete Board"
        />
        <input
          name="editBoard"
          onClick={[Function]}
          type="button"
          value="Edit Board"
        />
      </li>
      <li
        key="2"
      >
        <span
          onClick={[Function]}
        >
          <input
            disabled={true}
            id="kd5o3ve2858vpjxd7wjp"
            name="boardName"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="text"
            value="Sports Analyst Cambridgeshire"
          />
        </span>
        <input
          name="deleteBoard"
          onClick={[Function]}
          type="button"
          value="Delete Board"
        />
        <input
          name="editBoard"
          onClick={[Function]}
          type="button"
          value="Edit Board"
        />
      </li>
    </ul>
    <input
      name="addBoard"
      onClick={[Function]}
      type="button"
      value="Add Board"
    />
  </div>
</Boards>
`;

exports[`Boards component update board snapshot 2`] = `
<Boards
  boards={Array []}
  updateBoard={[Function]}
>
  <div>
    <ul>
      <li
        key="0"
      >
        <span
          onClick={[Function]}
        >
          <input
            disabled={true}
            id="2skdde39hzdo08aqcdgd"
            name="boardName"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="text"
            value="testing"
          />
        </span>
        <input
          name="deleteBoard"
          onClick={[Function]}
          type="button"
          value="Delete Board"
        />
        <input
          name="editBoard"
          onClick={[Function]}
          type="button"
          value="Edit Board"
        />
      </li>
      <li
        key="1"
      >
        <span
          onClick={[Function]}
        >
          <input
            disabled={true}
            id="14w43d4opubteu021ato"
            name="boardName"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="text"
            value="Bedfordshire Illinois Liaison"
          />
        </span>
        <input
          name="deleteBoard"
          onClick={[Function]}
          type="button"
          value="Delete Board"
        />
        <input
          name="editBoard"
          onClick={[Function]}
          type="button"
          value="Edit Board"
        />
      </li>
      <li
        key="2"
      >
        <span
          onClick={[Function]}
        >
          <input
            disabled={true}
            id="kd5o3ve2858vpjxd7wjp"
            name="boardName"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="text"
            value="Sports Analyst Cambridgeshire"
          />
        </span>
        <input
          name="deleteBoard"
          onClick={[Function]}
          type="button"
          value="Delete Board"
        />
        <input
          name="editBoard"
          onClick={[Function]}
          type="button"
          value="Edit Board"
        />
      </li>
    </ul>
    <input
      name="addBoard"
      onClick={[Function]}
      type="button"
      value="Add Board"
    />
  </div>
</Boards>
`;
